### PR TITLE
[IGA] Python geometry

### PR DESCRIPTION
**Pull request by oberbichler**
- Extend the python interface
- Some clean-up...

### CODE REVIEW

File: applications/IgaApplication/IgaApplication.py [hunk 1]
```diff
@@ -1,14 +1,12 @@
...
-#          |   /   \  ___  /  /_\  \
-#          |   \    \_\  \/    |    \
-#          |___|\______  /\____|__  /
-#                      \/         \/  Application
-#
-#   License: BSD License
-#            Kratos default license: kratos/license.txt
-#
+# 
+#   KRATOS  _____________
```
Comment (philbucher) [inline applications/IgaApplication/IgaApplication.py]: Sorry but this style is exclusively reserved by the MappingApplication ;)

I will take a look later

**Review by philbucher — COMMENTED** [review]
Ok from my side, approval I would leave to @tteschemacher

**Comment by oberbichler** [comment]
AFAIK he still works in his own branch and has disabled github notifications. I would be happy if we can merge this as soon as possible

File: applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp [hunk 2]
```diff
@@ -21,203 +19,802 @@
...
-        ;
-    }
+void RegisterPoint1D(
+    pybind11::module& m,
+    const std::string& name)
+{
+    namespace py = pybind11;
+    using namespace pybind11::literals;
+
+    using Type = Kratos::array_1d<double, 1>;
```
Comment (RiccardoRossi) [inline applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp:37]: calling it "type" is quite generic. Can't u call IntervalType or something of the sort?

i am speaking thinking of future naming conflicts (i do appreciate that Type is local to the function, so ignore this comment if you really dislike it)

**Review by RiccardoRossi — APPROVED** [review]

**Comment by RiccardoRossi** [comment]
@tteschemacher just please take a look at this to be aware of the changes.

@oberbichler don't forget to delete the branch

File: applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp (hunk 2 above)
Comment (oberbichler) [reply applications/IgaApplication/custom_python/add_custom_utilities_to_python.cpp:37]: The idea is that there is only one `Type` alias per `Register<Type>` function.

But there is the risk of naming conflicts. Types like `Interval` or `Curve2D` might be also declared in other applications. Not at the moment but maybe in the future. So we should think about using namespaces, using prefixes, moving stuff into the core,...

**Comment by oberbichler** [comment]
@RiccardoRossi Thank`s! I already talked with tteschemacher before the merge.

**Pull request merged**